src/main/minio.js

```diff
@@ -304,7 +305,8 @@ export default class Client {
       options.headers['content-length'] = payload.length
     }
     var sha256sum = ''
-    if (!this.anonymous) sha256sum = Crypto.createHash('sha256').update(payload).digest('hex')
+    var enableSHA256 = !(this.anonymous || this.secure)
+    if (enableSHA256) sha256sum = Crypto.createHash('sha256').update(payload).digest('hex')
```

Comment:
why not make this variable as part of 'this.enablePayloadCksum'  and use it everywhere. Just that we can avoid having `enableSha256` lying everywhere.